packages/alfa-rules/src/sia-er62/diagnostics.ts

```diff
@@ -0,0 +1,202 @@
+import { Diagnostic } from "@siteimprove/alfa-act";
+import { Array } from "@siteimprove/alfa-array";
+import { Device } from "@siteimprove/alfa-device";
+import { Element } from "@siteimprove/alfa-dom";
+import { Equatable } from "@siteimprove/alfa-equatable";
+import { Hash, Hashable } from "@siteimprove/alfa-hash";
+import { Serializable } from "@siteimprove/alfa-json";
+import { Map } from "@siteimprove/alfa-map";
+import { Result } from "@siteimprove/alfa-result";
+import { Context } from "@siteimprove/alfa-selector";
+import { Sequence } from "@siteimprove/alfa-sequence";
+import { Property, Style } from "@siteimprove/alfa-style";
+
+import * as json from "@siteimprove/alfa-json";
+
+import { Contrast } from "../../src/common/diagnostic/contrast";
+
+import { Serialise } from "./serialise";
+
+type Name = Property.Name | Property.Shorthand.Name;
+
+export class ElementDistinguishable
+  implements Equatable, Hashable, Serializable
+{
+  public static of(
+    style: Iterable<readonly [Name, string]> = [],
+    pairings: ReadonlyArray<Contrast.Pairing> = []
+  ): ElementDistinguishable {
+    return new ElementDistinguishable(Map.from(style), pairings);
```

Comment:
At this point, we can be a bit more generous in what we accept
```suggestion
    pairings: Iterable<Contrast.Pairing> = []
  ): ElementDistinguishable {
    return new ElementDistinguishable(Map.from(style), Array.from(pairings));
```